cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -0,0 +1,365 @@
+package resource
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progressEvents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/openlyinc/pointy"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+var RequiredFields = []string{constants.PubKey, constants.PvtKey, constants.ClusterName, constants.ProjectID}
+
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup() // logger setup
+
+	_, _ = logger.Debugf("Create snapshot for Request() currentModel:%+v", currentModel)
+
+	// validate the request
+	event, client, err := ValidateRequest(currentModel)
```

Comment:
The only responsibility of `ValidateRequest` should be to validate the request. It should not return the client.